inc/EpollWrapper.hpp

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include "Log.hpp"
+
+#include <cerrno>
+#include <cstring>
+#include <stdexcept>
+#include <sys/epoll.h>
+#include <unistd.h>
+#include <vector>
+
+/**
+ * @brief Wrapper class for epoll instance.
+ *
+ * This class is a wrapper for the epoll instance. It is responsible for the creation and destruction
+ * of the epoll instance. It also provides a function to wait for events on the epoll instance and functions to add,
+ * modify and remove file descriptors from the epoll instance.
+ * It can also be mocked for testing purposes.
+ */
+class EpollWrapper {
+
+private:
+	static const int s_epollMaxEvents = 10; /**< Default maximum number of events for epoll instance */
+	static const int s_epollTimeout = 1000; /**< Default timeout for epoll instance in miliseconds */
+
+public:
+	explicit EpollWrapper(size_t maxEvents = s_epollMaxEvents, int epollTimeout = s_epollTimeout);
+	virtual ~EpollWrapper();
+
+	virtual int waitForEvents();
```

Comment:
Yes: the Mock version inherits from EpollWrapper, it needs to be able to overwrite the function calls from the base version. This way a MockEpollWrapper can be passed as EpollWrapper (since it inherits from it) and the functions to base call the overridden versions.

Interestingly enough the gMock Cookbook advises against this style: https://google.github.io/googletest/gmock_cook_book.html#alternative-to-mocking-concrete-classes

They recommend defining an interface, code to that interface and mock the interface. 
This would mean that we have IEpollWrapper (interface), EpollWrapper (concrete class), MockEpollWrapper (mock implementing IEpollWrapper). 
They argue that 
- making something virtual creates an extension point where subclasses can tweak the class’ behavior. 
- mocking concrete classes directly is problematic as it creates a tight coupling between the class and the tests

This seems like much overhead, for the benefits:
- I don't think that we would make a class that inherits from these classes
- Tight coupling is not that bad, since we only test for that specific implementation

It would make more sense if we had an interface IMultiplexer, which can be implemented as EpollMultiplexer, SelectMultiplexer and MockMultiplexer. 
But I think this would generate too much complexity for this project > maybe if we want to continue on this server afterwards?

Reply:
Ok, that makes sense. In the scope of our project we won't need the interface implementation as we will not be creating any subclasses of the EpollWrapper. If we were to work on the server later on, we can still create a branch for this  and change the implementation. For the current goal of validating the project asap the current state is good as is.